styleframe/style_frame.py

```diff
@@ -382,12 +397,7 @@ def to_excel(self, excel_writer: Union[str, pd.ExcelWriter, pathlib.Path] = 'out
             if excel_writer.engine != 'openpyxl':
                 raise TypeError('styleframe supports only openpyxl, attempted to use {}'.format(excel_writer.engine))
 
-        # dealing with needed pandas.to_excel defaults
-        header = kwargs.pop('header', True)
-        index = kwargs.pop('index', False)
-        startcol = kwargs.pop('startcol', 0)
-        startrow = kwargs.pop('startrow', 0)
-        na_rep = kwargs.pop('na_rep', '')
+        header, startcol, startrow, na_rep = StyleFrame._to_excel_pandas_defaults(kwargs)
```

Comment:
I suggest to use `self` to properly support subclassing.

```suggestion
        header, startcol, startrow, na_rep = self._to_excel_pandas_defaults(kwargs)
```